skademlia/client_filter.go

```diff
@@ -0,0 +1,25 @@
+package skademlia
+
+import (
+	"time"
+)
+
+type ClientFilter func([]*ID) []*ID
+
+func Dummy(ids []*ID) []*ID {
+	return ids
+}
+
+func FilterFailedAgo(ago time.Duration) ClientFilter {
```

Comment:
Any reason why on having it time-based vs attaching a boolean to an ID like `closed bool`?

Reply:
but how bool help us to call this node later?
current implementation will skip node for some time after a failure instead of trying to connect to it immediately after one

Reply:
Hmm, using time though would be quite a hacky fix - this means after some period of time, it is still possible for the node to reconnect right? Is this the root of the data race?

Reply:
thats only to save some time trying to connect to failing node. ideally this time delay should be exponential depending on number of consecutive failed calls